google/auth/_default.py

```diff
@@ -17,14 +17,20 @@
 Implements application default credentials and project ID detection.
 """
 
+from collections.abc import Sequence
 import io
 import json
 import logging
 import os
+from typing import Optional, TYPE_CHECKING
 import warnings
 
 from google.auth import environment_vars
 from google.auth import exceptions
+
+if TYPE_CHECKING:  # pragma: NO COVER
+    from google.auth.credentials import Credentials
+    from google.auth.transport import Request
```

Comment:
```suggestion
if TYPE_CHECKING:  # pragma: NO COVER
    # flake8 raises an error because these lines are not used in the code,
    # they are only used for type-hinting, hence the F401 pragma. 
    from google.auth.credentials import Credentials  # noqa: F401
    from google.auth.transport import Request  # noqa: F401
```